### DISCUSSION TITLE
Extracting the Order of Reflection of a Path

### DISCUSSION THREAD

**Discussion by steffenitely** [NVlabs/sionna#681] [Q&A] [answered]
Hi everyone,

I was wondering if there's an efficient way to extract the order of reflections for a given path. Specifically, I'm looking for a method to identify the reflection order without needing to enumerate through each depth level until the maximum depth and then subtract previously found paths.

Is there an existing approach that simplifies this process, or is it necessary to iterate through all the possible paths to determine the reflection order?

Any insights would be appreciated!

Thanks in advance!

**Comment by faycalaa** [comment, marked as answer]
Hi @steffenitely,

I am not sure to have correctly understood you question, but if you are looking for the length of a path (i.e., number of times a path interacts with the scene), you could use the  [Paths.objects](https://github.com/NVlabs/sionna/blob/0aeca346148f1e47a4303f391da1db07e6df1ea7/sionna/rt/paths.py#L857) variable.

It has shape `[max_depth, num_targets, num_sources, max_num_paths]` and stores the indices of the intersected objects, or -1 if no object was intersected, which means that the path has ended. You could extract from it the 1D vector of size `[max_depth]` corresponding to the path and link of interest, and then look at the highest index storing a valid intersected objects (i.e., with index not equal to -1). That should correspond to the path length.

**Reply by steffenitely** [reply]
Hi @faycalaa 
that seems like an easier way of extracting the information of how many times a path has reflected within the scene
thanks for the quick answer!
